lib/streamlit/string_util.py

```diff
@@ -67,6 +73,46 @@ def validate_emoji(maybe_emoji: str | None) -> str:
         )
 
 
+def validate_material_icon(maybe_material_icon: str | None) -> str:
+
+    supported_icon_packs = [
+        "material",
+    ]
+
+    if maybe_material_icon is None:
+        return ""
+
+    icon_regex = r"^\s*:(.*)\/(.*):\s*$"
+    icon_match = re.match(icon_regex, maybe_material_icon)
+
+    if not icon_match:
+        raise StreamlitAPIException(
+            f'The value `"{maybe_material_icon}"` is not a valid Material icon. '
+            f"Please use a Material icon shortcode like **`:material/thumb_up:`**"
+        )
+    pack_name, icon_name = icon_match.groups()
+
+    if pack_name not in supported_icon_packs:
+        icon_packs_repr = ", ".join(f"`{pack}`" for pack in supported_icon_packs)
+        raise StreamlitAPIException(
+            f'The value `"{maybe_material_icon}"` is not a valid Material icon.'
+            f" Please use a Material icon shortcode like **`:material/thumb_up:`**. "
+            f"Only {icon_packs_repr} icon packs are supported."
```

Comment:
nit: will probably run into this while generalizing the error message, but it'd be slightly better to move the space onto this line so that a shared error message an be used everywhere else and appended to here